scripts/version.go

```diff
@@ -0,0 +1,193 @@
+		logAndExit(err)
+
+		fmt.Println(out.String())
+	}
+}
+
+// runCmd runs the given command and returns the combined output and/or an error
+func runCmd(c string) (string, error) {
+	parts := strings.Split(c, " ")
+	//nolint:gosec
+	cmd := exec.Command(parts[0], parts[1:]...)
+	out, err := cmd.CombinedOutput()
+	return strings.TrimSpace(string(out)), err
+}
+
+// version returns the correct version string to use given the current state of
+// this repo
+func version() (*semver.Version, error) {
+	desc, err := runCmd("git describe --always --tags --dirty")
+	if err != nil {
+		return nil, fmt.Errorf("git describe failed: %w", err)
+	}
+
+	return parseVersion(desc)
+}
+
+// parseVersion parses a version string as returned by 'git describe --always',
+// and if it's not on a tag, increments the patch so it's a valid prerelease
+func parseVersion(in string) (*semver.Version, error) {
+	ver, err := semver.NewVersion(in)
+	if err != nil {
+		return nil, err
+	}
+
+	// if we're not on a tag, use the bits that git appended to the tag as the
+	// prerelease, but first increment the patch
+	// no need to check error since we already know this is an integer
+	v2 := *ver
+	if v2.Prerelease() != "" {
+		// one call just removes the prerelease part, so we need a second call
+		v2 = v2.IncPatch().IncPatch()
+	} else if v2.Metadata() != "" {
+		// if only metadata is provided, IncPatch() increments the patch and
+		// removes the metadata part
+		v2 = v2.IncPatch()
+	}
+
+	// no-op on releases
+	v2, _ = v2.SetPrerelease(ver.Prerelease())
+	v2, _ = v2.SetMetadata(ver.Metadata())
+
+	return &v2, nil
+}
+
+func release(ver *semver.Version, nosign bool) error {
+	// make sure this is being run from main
+	out, err := runCmd("git status -sb -u no")
+	if err != nil {
+		return err
+	}
+
+	if strings.TrimSpace(out) != "## main...origin/main" {
+		return fmt.Errorf("must release from the main branch - check it out and pull first")
+	}
+
+	// make sure we're up to date
+	out, err = runCmd("git pull")
+	if err != nil {
+		return err
+	}
+
+	if strings.TrimSpace(out) != "Already up to date." {
+		fmt.Printf("git pull:\n%s\n", out)
+	}
+
+	nextVer := ver
+	*nextVer = ver.IncPatch()
+
+	// default to GPG-signed tags, unless -nosign was set
+	tagArgs := "-sm"
+	if nosign {
+		tagArgs = "-am"
+	}
+
+	cmd := fmt.Sprintf(`git tag %s "v%s" v%s`, tagArgs, nextVer, nextVer)
+	out, err = runCmd(cmd)
+	fmt.Println(out)
+	if err != nil {
+		return fmt.Errorf("failed to run cmd %s: %w", cmd, err)
+	}
+
+	cmd = fmt.Sprintf(`git push origin v%s`, nextVer)
+	out, err = runCmd(cmd)
+	fmt.Println(out)
+	if err != nil {
+		return fmt.Errorf("failed to run cmd %s: %w", cmd, err)
+	}
+
+	fmt.Printf("v%s is tagged and pushed!\n", nextVer)
+
+	ghReleaseCmd := fmt.Sprintf("gh release create v%s --generate-notes", nextVer)
+	out, err = runCmd(ghReleaseCmd)
+	fmt.Println(out)
+	if err != nil {
+		if strings.Contains(err.Error(), `"gh": executable file not found in $PATH`) {
```

Comment:
Could we check `gh` is installed at the beginning of this function?

Reply:
probably - this was copy/pasted from elsewhere though so I should probably do that in the source first